src/olympia/devhub/tasks.py

```diff
@@ -414,20 +415,44 @@ def run_addons_linter(path, listed=True):
     if not listed:
         args.append('--self-hosted')
 
+    stdout, stderr = tempfile.TemporaryFile(), tempfile.TemporaryFile()
+
     process = subprocess.Popen(
-        args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        args,
+        stdout=stdout,
+        stderr=stderr,
+        # default but explicitly set to make sure we don't open a shell.
+        shell=False,
+        # Avoid buffering to make sure we don't accidently trim some output.
+        bufsize=0
```

Comment:
I rewrote it a bit and it handles now both, in-memory and temporary files just like the django upload handler does. For simplicity reasons I'm using the same configuration here and hope it helps.

The thing is, the actual output and the file size don't have to be related but it's an indice. What do you think?